### PR TITLE
Fix notify dto

### PR DESCRIPTION
**Pull request by fleupold**
# Description
Notification calls from the driver to solver engines are currently failing due to:

> Failed to deserialize the JSON body into the target type: invalid type: map, expected variant identifier at line 1 column 143

I broke this with #2192 where I missed adding the flatten directive on the driver dto.

# Changes
- [x] Fix dto on both sides (we can no longer deny_unknow_fields due to this [serde bug](https://github.com/serde-rs/serde/issues/1600))
- [x] Warn if the notify call is not successful

## How to test
Run e.g. the partial fill colocation test case. See warning before adjusting the dto and now no longer see it.

### CODE REVIEW

File: crates/solvers/Cargo.toml [hunk 1]
```diff
@@ -36,7 +36,7 @@ thiserror = "1"
 tokio = { version = "1", features = ["macros", "rt-multi-thread", "signal", "time"] }
 toml = "0.7"
 tower = "0.4"
-tower-http = { version = "0.4", features = ["trace"] }
+tower-http = { version = "0.4", features = ["limit", "trace"] }
```
Comment (fleupold) [inline crates/solvers/Cargo.toml:39]: slightly unrelated trivial change needed to make the solver binary compile in isolation.

**Review by ahhda — APPROVED** [review]

**Review by squadgazzz — APPROVED** [review]
